empress/vistrans/ReconRenderer.py

```diff
@@ -0,0 +1,418 @@
+import RenderClasses
+import tkinter as tk
+
+
+NODE_RADIUS = 5
+TREE_OFFSET = [20, 30]
+PARASITE_TREE_OFFSET = 15
+HOST_TREE_COLOR = 'green'
+PARASITE_TREE_COLOR = 'red'
+WINDOW_WIDTH = 1000
+WINDOW_HEIGHT = 1000
+WINDOW_NAME = "reconciliation"
+
+
+def computeHostNodeLogicalPositions(host_tree):
+    """ 
+    Sets the logicalRow and logicalCol values of each Node in host_tree.
+    Assumes that each host Node has its order set already and this function
+    uses those values and structure of the tree to set the logicalRow and logicalCol
+    :param host_tree:  A Tree object representing the host tree
+    :return None
+    """
+    #set logical cols
+    for node in host_tree.allNodes:
+        node.logicalCol = node.order
+
+    #sets logical row values for leaves in the order they appear in host_tree.leaves
+    logical_row_counter = 0
+    for leaf in host_tree.leaves:
+        leaf.logicalRow = logical_row_counter
+        logical_row_counter += 1
+    
+    #helper function to assign row values, postorder traversal
+    computetHostNodeLogicalPositions_helper(host_tree.rootNode)
+
+    
+def computetHostNodeLogicalPositions_helper(node):
+    """takes a Node, and calculates logical row values"""
+
+    #if both children of node have a logical row value, we can calculate the logical row value of node
+    if node.rightChild.logicalRow is not None and node.leftChild.logicalRow is not None:
+        node.logicalRow = ((node.rightChild.logicalRow+node.leftChild.logicalRow)/2)
+        return
+
+    #recursively calculate logical row values of the right subtree 
+    if node.rightChild.logicalRow == None:
+        computetHostNodeLogicalPositions_helper(node.rightChild)
+    #recursively calculate logical row values of the left subtree
+    if node.leftChild.logicalRow == None:
+        computetHostNodeLogicalPositions_helper(node.leftChild)
+    
+    #finally, calculate logical row value of node using just-calculated children values
+    node.logicalRow = ((node.rightChild.logicalRow+node.leftChild.logicalRow)/2)
+
+
+    ### Here's what needs to happen here:
+    ### The logicalCol of each Node is just its order.
+    ### The logicalRows are computed as follows:
+    ###     1.  For leaves, we iterate through the list of leaves for the tree from left (top)
+    ###         to right (bottom) and make the logicalRow the indices 0 to numLeaves-1
+    ###     2.  Moving up the tree, the logical row of a node is the midpoint (a float) of 
+    ###         the logical rows of its two children.
+
+
+def computeParasiteNodeLogicalPositions(parasite_tree, host_tree, recon_map):
```

Comment:
Please use snake case function names.

Reply:
this is our old code, we're just keeping it here for reference as we write new code that is compatible with the new data structures

Reply:
In that case, keep in in your computer but don't push it to the codebase.